app/main.py

```diff
@@ -1,2 +1,90 @@
+from __future__ import annotations
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.cells = [[] for i in range(8)]
+        self.next_index = 0
+        self.length = 0
+
+    def __get_index__(self, key: Hashable, action: str) -> int:
+        cell_index = hash(key) % len(self.cells)
+        is_key_exists = False
+
+        def find_key(sliced_list: list) -> int:
+            nonlocal is_key_exists
+            for index, items in enumerate(sliced_list):
+                if action == "add":
+                    if not items or items[0] == key:
+                        return index
+                elif action == "get":
+                    if items and items[0] == key:
+                        is_key_exists = True
+                        return index
+
+        _index = find_key(self.cells[cell_index:])
+        if _index is None:
+            _index = find_key(self.cells[:cell_index])
+        else:
+            _index += len(self.cells[:cell_index])
+        if action == "get" and not is_key_exists:
+            raise KeyError(f"No such a key {key}")
+        return _index
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        cell_index = self.__get_index__(key, "add")
+        cell_content = self.cells[cell_index]
+        if not cell_content:
+            self.length += 1
+        self.cells[cell_index] = [key, value]
+        if len(self) > len(self.cells) / 3 * 2:
+            self.resize()
+
+    def resize(self) -> None:
+        items_present = {
+            items[0] : items[1] for items in self.cells if items
+        }
```

Comment:
Do not use built-in `dict` for this task as mentioned in the task description. Besides that, `items[0]: items[1]` is not descriptive enough, use unpacking and variable assignment to improve the readability.